src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/alignment/ContigAlignmentsModifier.java

```diff
@@ -208,18 +208,46 @@ private static AlignmentInterval clipAlignmentInterval(final AlignmentInterval i
                 refBasesConsumed += ce.getLength();
             }
         }
+        if (newMiddleSection.size() < 2)
+            throw new GATKException("After clipping, the middle section of an alignment isn't split, which seems wrong\t" + input.toPackedString() + "\tclip length " +
+                    clipLengthOnRead + "\tclip from end " + (clipFrom3PrimeEnd? "3":"5"));
+
+        // guard against edge case where the requested clipping clip away a whole alignment block that neighbors an indel operation
+        // e.g. "30M5I..." and user requests 30 bases to be chipped away hence leading to "...S5I", which we consider to be invalid
+        final CigarElement firstOp = newMiddleSection.get(0);
+        final CigarElement secondOp = newMiddleSection.get(1);
+        if ( firstOp.getOperator().equals(CigarOperator.S) && secondOp.getOperator().isIndel() ) {
+            // the input cigar, if good, cannot have neighboring INS/DEL operations so we are safe here
```

Comment:
I'm confused by this comment, we don't seem safe because you immediately check for a an error condition after this.

Reply:
the doc is outdated, so removed
I initially thought I would need the next operation as an anchor, which must be an alignment in this `if` block. But then I decided the anchoring alignment is unnecessary.